### ISSUE TITLE
Evaluate story generation language model?

### ISSUE THREAD

**Issue by edbltn** [pytorch/fairseq#368]
I am looking to assess a story generation language model according to the automatic evaluation metrics described in the Fan et al. 2018 paper, namely perplexity and prompt ranking accuracy.

I tried running `eval_lm.py` on the fusion checkpoint provided in the readme (`fusion_checkpoint.pt`) using the following command:
```
python eval_lm.py data-bin/writingPrompts --path 'data-bin/models/fusion_checkpoint.pt'
```
When I first tried, I got an error telling me there was no `dict.txt` file. So I tried renaming one of the `wp_source.dict.txt` and `wp_target.dict.txt` files to `dict.txt`. When I tried running again I got:
```
OSError: Model file not found: /checkpoint/angelafan/wp_open_source/checkpoint_best.pt
```
Likewise, when I tried to do it on a story generation language model trained on a dataset of my own, I got a `NotImplementedError`

It seems that `eval_lm.py` is not designed for evaluating language generation models that require two dictionaries, source and target, but instead is designed for simple language models.

My question is: how can I evaluate the story generation model in terms of perplexity? In addition, is there any code provided in the fairseq repository to measure prompt ranking accuracy?

**Comment by ruidongtd** [comment]
`python eval_lm.py data-bin/writingPrompts \
--path fusion_checkpoint.pt \
--model-overrides "{'pretrained_checkpoint':'pretrained_checkpoint.pt'}" \
--task translation --max-sentences 8`

error:

`eval_lm.py: error: unrecognized arguments: --model-overrides {'pretrained_checkpoint':'pretrained_checkpoint.pt'}`

**Comment by huihuifan** [comment]
Sorry I missed this. For perplexity, I reported the perplexity numbers from NLL loss at training time (the test set PPL corresponding to the best validation set PPL). For prompt ranking accuracy- we do not have this functionality officially bundled with fairseq, sorry. For each story, I sampled 9 unrelated prompts and assessed the model likelihood under those 9 and the true prompt, then averaged across all stories in the test set.
